File: modules/algorithms-and-ds/modules/data-structures/linear/lists/src/python/singly-linked-list/driver_printListFromHead.py
from SinglyLinkedList import SinglyLinkedList
from Node import Node

print("####### ( Print Node values from the 'head' until 'tail' ) ######")

sll = SinglyLinkedList()

sll.printListFromHead()

sll.head = Node(40)
sll.printListFromHead()

sll.head.next = Node(50)
sll.printListFromHead()

sll.head.next.next = Node(60)
sll.printListFromHead()

File: modules/old-studies/old-algorithms-and-ds/modules/data-structures/linear/lists/src/cpp/singly-linked-list/driver_printListFromHead.cpp
#include "SinglyLinkedList.h"
#include <iostream>

int main()
{
    SinglyLinkedList list;
    list.printListFromHead();

    list.head = new Node(10);
    list.printListFromHead();

    list.head->next = new Node(20);
    list.printListFromHead();

    list.head->next->next = new Node(30);
    list.printListFromHead();

    return 0;
}
